### ISSUE TITLE
When you have exactly 3 children in nested flexible it fails to save due to clash with isFlexibleStructure

### ISSUE THREAD

**Issue by jackabox** [whitecube/nova-flexible-content#56]
Hey All,

Tried to save an element having 3 nested children, which has a key total of **3**. In the ScopedRequest.php you're checking against the count of the keys and then doing some work off that to get the nested files. With having exactly 3 child items, it seems to throw an issue here. 

Not sure if we can detect anything else? Having 2/4 seems to work. I'm guessing this is due to the default structure you have of 

```
[
  'layout' => '',
  'key' => '',
  'attributes' => ''
]
```

The error being returned in the log is

```
[message]
Undefined index: key {"userId":1,"exception":"[object] (ErrorException(code: 0): Undefined index: key at /Users/jack/Sites/travelteer/vendor/whitecube/nova-flexible-content/src/Http/ScopedRequest.php:95)
```

Using **v0.1.9**.

**Comment by gwleuverink** [comment]
I get the same error when using the `limit()` fluent method. 

When you reach the maximum of flexible fields defined by `limit` the resource can't be created/updated and throws an exception with the message `Undefined index: key`.

@jackabox Are you per chance also using the `limit` method? If not I'll open a new issue.

**Comment by jackabox** [comment]
Not using the limit on this one

**Comment by gwleuverink** [comment]
I was wrong about the limit method. I also get the error even without. Any help whit this is greatly appreciated!

**Comment by Nyratas** [comment]
This has been fixed in the current v0.1.10 release.
